hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/RecordIterators.java

```diff
@@ -80,12 +104,42 @@ public static ClosableIterator<RowData> getParquetRecordIterator(
               batchSize,
               path,
               splitStart,
-              splitLength));
+              splitLength,
+              filterPredicate,
+              recordFilter));
       if (castProjection.isPresent()) {
         return new SchemaEvolvedRecordIterator(itr, castProjection.get());
       } else {
         return itr;
       }
     }
   }
+
+  private static FilterPredicate getFilterPredicate(Configuration configuration) {
+    try {
+      return SerializationUtil.readObjectFromConfAsBase64(FILTER_PREDICATE, configuration);
+    } catch (IOException e) {
```

Comment:
Not sure how the user can config the predicate in configuration, and what the `UNBOUND_RECORD_FILTER` is used for ? Because Hudi may has merging logic of base parquet file and logs files, does these filtering utilities only work for parquets?

Reply:
@danny0405, the filters could be passed to Hadoop's configuration entries prefixed with `FILTER_PREDICATE` that is `parquet.private.read.filter.predicate` for `HoodieTableSource#getParquetConf` and used by one of the available readers, `VectorizedParquetRecordReader` or `ParquetRecordReader`. Meanwhile, `UNBOUND_RECORD_FILTER` which is `parquet.read.filter` is used for native parquet read filter cofiguration.